src/main/scala/optics/poly/Focus.scala

```diff
@@ -188,9 +188,9 @@ object Focus {
 
     def composeLensTerms(lens1: Term, lens2: Term): FocusResult[Term] = {
       (lens1.tpe.asType, lens2.tpe.asType) match {
-        case ('[EOptional[err1, from1, to1]], '[EOptional[err2, from2, to2]]) => 
+        case ('[Optional[from1, to1]], '[Optional[from2, to2]]) =>
```

Comment:
Thanks for the help. It seems that we will need to define all the potential combination here (Lens -> Lens), (Lens -> Lens), (Lens -> Optional). It would be better if optics inherited from another but we don't have that yet in Monocle 2.